### PR TITLE
Helper function to reload a quantized state_dict

### PR DESCRIPTION
**Pull request by ManoBharathi93**
issue number: #162

### CODE REVIEW

**Comment by dacorvo** [comment]
@ManoBharathi93 your pull-request is not exactly aligned with the [contributions guidelines](https://github.com/huggingface/quanto/blob/main/CONTRIBUTING.md#create-a-pull-request).
In particular, please try to use conventional commits.
Regarding your tests, you simply duplicated the existing quantize tests without calling the newly introduced function, hence not verifying it works.

What I suggest for requantize tests is that you simply extract `test_serialize_quantized_mlp` from `test_quantize_mlp.py` and put it in a new `test_requantize_mlp.py` file.
Then, change the two lines at 123-124 to use your new helper.

**Comment by ManoBharathi93** [comment]
replacing  line 123-124 code with  `requantize(model,state_dict)`

**Comment by ManoBharathi93** [comment]
@dacorvo , can you check my changes please.

**Comment by dacorvo** [comment]
Please rebase your branch, then make sure it follows the contributing guidelines: the CI fails even before running the tests.
If you have trouble rebasing your branch, please look at the instructions in the guidenies and let me know if something is not clear.

**Comment by dacorvo** [comment]
Also, as I said before please do **not** cc all the test from `test_quantize_mlp.py` in your new test file. This increases the codebase without any benefit.

**Comment by ManoBharathi93** [comment]
> Also, as I said before please do **not** cc all the tests from `test_quantize_mlp.py` in your new test file. This increases the codebase without any benefit.

Okay I will modify in the new commit

**Comment by ManoBharathi93** [comment]
@dacorvo, Added test for MLP and serialize_quantized_mlp only and removed rest of the tests
image

**Comment by dacorvo** [comment]
Commit messages:
  🚩 added_requantize
  🚩 Add: Helper to reload a quantized state_dict
  🚩 add:helper to requantize
  ✅ fix:style corrections
Error: 🚫 According to the conventional-commits specification, some of the commit messages are not valid.

**Comment by dacorvo** [comment]
If you just want to reword your commits, rebase iteratively from the root of your branch:

```
$ git rebase -i c75cd06
```

In the editor, select r or reword in front of the guilty commits, save and exit.
Now, rewrite each message when prompted.
Once done:

```
$ git push -f origin reload_quantized_state_dict
```

**Comment by ManoBharathi93** [comment]
> Commit messages: 🚩 added_requantize 🚩 Add: Helper to reload a quantized state_dict 🚩 add:helper to requantize ✅ fix:style corrections Error: 🚫 According to the conventional-commits specification, some of the commit messages are not valid.

It means, red flagged commits needs to rename as fix: added_requantize,... ?

**Comment by dacorvo** [comment]
They should all start with an allowed prefix, whose list is here:
https://github.com/huggingface/quanto/commit/416db04d7eab6d5af4295864f698cf5bcb85443c

**Comment by dacorvo** [comment]
Alternatively, you can squash all your commits into one clean commit an rephrase it to `feat: add requantize helper` or something like that (just select s on the left while rebasing).

**Comment by ManoBharathi93** [comment]
> They should all start with an allowed prefix, whose list is here: [416db04](https://github.com/huggingface/quanto/commit/416db04d7eab6d5af4295864f698cf5bcb85443c)

Then in my case all should be prefix with fix: ..
image

**Comment by ManoBharathi93** [comment]
> feat: add requantize helpe

Yes I was my first time  with this git features. I was squash as you said, but  it deletes all my commits and my head points to this commit `c75cd` now and 
image

**Comment by dacorvo** [comment]
```
$ git rebase --continue
```

**Comment by ManoBharathi93** [comment]
> git rebase --continue

git rebase --continue
error: cannot 'squash' without a previous commit
error: please fix this using 'git rebase --edit-todo'.

**Comment by dacorvo** [comment]
ah yes, the first one must stay if you squash.
You can alwaus git rebase --abort and restart if you want

**Comment by ManoBharathi93** [comment]
@dacorvo , Thanks for your patience and helping me, it takes some time to understand for me..

File: test/model/test_requantize_mlp.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+        torch.save(state_dict, b)
+        b.seek(0)
+        weights_only = serialization == "weights_only"
+        return torch.load(b, weights_only=weights_only)
+
+
+@pytest.mark.parametrize("weights", [qint8], ids=["w-qint8"])
+@pytest.mark.parametrize("dtype", [torch.float16, torch.float32], ids=["fp16", "fp32"])
+@pytest.mark.parametrize("serialization", ["weights_only", "pickle", "safetensors"])
+@pytest.mark.skip_device("cuda")
```
Comment (dacorvo) [inline test/model/test_requantize_mlp.py:66]: Please do not skip the test on CUDA

Comment (ManoBharathi93) [reply test/model/test_requantize_mlp.py:66]: > Please do not skip the test on CUDA
I don't have cuda, that's the reason I skipped it.. I can remove that line and make a PR, but it fails locally in my laptop

Comment (ManoBharathi93) [reply test/model/test_requantize_mlp.py]: Removed skip test line and made a PR again

Comment (ManoBharathi93) [reply test/model/test_requantize_mlp.py]: image
@dacorvo  Thanks, Now I understand rebase feature. Kindly  review this PR.

**Comment by ManoBharathi93** [comment]
@dacorvo do  I missed something ?

**Comment by dacorvo** [comment]
> @dacorvo do I missed something ? I have been waiting for your comments..test cases passed

Patience is the key ... you must understand that:

- this is an open-source project,
- I have many other things to do.

Also, please credit @calmitchell617 in your commit message because you basically cced his branch.

**Comment by ManoBharathi93** [comment]
Credits:@calmitchell617 with help of his work I made this PR  Thanks 😊

**Comment by ManoBharathi93** [comment]
@calmitchell617 work has to be merged since I reuse his code with some extra test case I am closing this pr.

**Comment by dacorvo** [comment]
@ManoBharathi93 thank you for acknowledging this. I have other issues that require help if you want to contribute.

**Comment by ManoBharathi93** [comment]
> @ManoBharathi93 thank you for acknowledging this. I have other issues that require help if you want to contribute.

😊 absolutely love to contribute.
